### PR TITLE
Well.hpp: forward WellConnections

### PR DESCRIPTION
**Pull request by akva2**
Reduces hotness of WellConnections.hpp by 82% (235 files, 51 instead of 286)

### CODE REVIEW

File: opm/input/eclipse/Schedule/Well/Well.hpp [hunk 1]
```diff
@@ -536,6 +536,7 @@ class Well {
     const std::string& groupName() const;
     Phase getPreferredPhase() const;
 
+    std::size_t getNoConnections() const;
```
Comment (bska) [inline opm/input/eclipse/Schedule/Well/Well.hpp:539]: Would you mind naming this `getNumConnections()` instead, please?  We might also/alternatively consider having a predicate of the form
```C++
bool haveConnections() const;
```
with an implementation along the lines of
```C++
bool haveConnections() const { return !this->connections.empty(); }
```
if we're **mostly** concerned about the case `getNoConnections() == 0`.

**Review by bska — COMMENTED** [review]
Looks good for the most part, but I think I'd prefer slightly different naming for the new member function.

File: opm/input/eclipse/Schedule/Well/Well.hpp (hunk 1 above)
Comment (akva2) [reply opm/input/eclipse/Schedule/Well/Well.hpp]: i went for bool hasConnections() - other methods in there use 'has' instead of 'have' already.

**Comment by akva2** [comment]
jenkins build this opm-simulators=4368 opm-grid=614 please

**Review by bska — APPROVED** [review]
Thanks a lot for the updates.  I'll merge this and its downstream companion PRs into the master branches.

**Pull request merged**